### ISSUE TITLE
Set replication_factor 1 to 2, rps is down 50%

### ISSUE THREAD

**Issue by kawhi-xl** [qdrant/qdrant#4392]
## Current Behavior

I set the collection replication_factor 1 to 2, rps is down 50%

## Steps to Reproduce

1. create a collection, insert some points.
2. make some request to search, rps may be 100.
3. create a new collection with 2 replication_factor, make some request to search, rps is only 50, down 50%!

## Expected Behavior

rps should not down, rps should be 100 same to the first collection.

## Possible Solution

I set the collections replication_factor with 2. I want the rps same with 1 replication_factor, which means my computing costs double!

## Possible Implementation

I read the source code, found the read_fan_out_factor parameter. I want every request only by one replicaset calculated. what should I do?

**Comment by kawhi-xl** [comment]
May it is not a bug, I want to know why read_fan_out can't set to the be requests calculated by only one replica.

**Comment by generall** [comment]
hey @kawhi-xl, how do you serve distributed deployment? It is multiple machines? How if there a load balancer?
`read_fan_out_factor` is only used in exceptional situations like ongoing updates, so it your case each request should land into one replica only

**Comment by kawhi-xl** [comment]
@generall yes, deployment with distributed, the cluster have 2 machines. read requests by grpc with slb, no write during read. What are the possible reasons? what shoud I do?

**Comment by kawhi-xl** [comment]
Also my search requests with no read_consistency.

**Comment by tellet-q** [comment]
@kawhi-xl could you please show how you calculate the rps?

**Comment by kawhi-xl** [comment]
@tellet-q  request with concurrency by client. rps = total_costs / requests_num. the colletion default_segment_number is 2, because my business requires high throughput.

**Comment by timvisee** [comment]
Could it be that your machines are very limited in terms of resources? What does your resource usage look like?

Also, how many points are we talking about? Are you sure they are properly indexed before running the test?

**Comment by kawhi-xl** [comment]
the cpu usage is 100%. about 4k+ points. the parm of indexing_threshold is 20000. Also test with 1 shared 6 replicas(6 node, 8C 16G of per node), the rps is similar with 1 shared 1 replica. the rps about 163.
![img_v3_02bh_3f2f99ac-51dc-4033-97c8-f6d2caeadffg](https://github.com/qdrant/qdrant/assets/41316350/754af33a-22ae-4310-8bbb-695f81a1ef46)
![rps](https://github.com/qdrant/qdrant/assets/41316350/30a9ae5f-0187-4898-8dd0-c493c37fce8b)
